plugins/tiddlywiki/markdown/markdown-it-tiddlywiki.js

```diff
@@ -0,0 +1,242 @@
+	var caption = slf.rules.footnote_caption(tokens,idx,options,env,slf);
+	var refid   = id;
+
+	if(tokens[idx].meta.subId > 0) {
+	  refid += ':' + tokens[idx].meta.subId;
+	}
+	return '<a class="footnote-ref" href=<<qualify "##fn' + id + '">> id=<<qualify "#fnref' + refid + '">>>' + caption + '</a>';
+};
+
+function render_footnote_open(tokens,idx,options,env,slf) {
+	var id = slf.rules.footnote_anchor_name(tokens,idx,options,env,slf);
+
+	if(tokens[idx].meta.subId > 0) {
+	  id += ':' + tokens[idx].meta.subId;
+	}
+	return '<li id=<<qualify "#fn' + id + '">>  class="footnote-item">';
+};
+
+function render_footnote_anchor(tokens,idx,options,env,slf) {
+	var id = slf.rules.footnote_anchor_name(tokens,idx,options,env,slf);
+
+	if(tokens[idx].meta.subId > 0) {
+	  id += ':' + tokens[idx].meta.subId;
+	}
+
+	// append variation selector to prevent display as Apple Emoji on iOS
+	return ' <a href=<<qualify "##fnref' + id + '">> class="footnote-backref">\u21A5\uFE0E</a>';
+};
+
+/// Add inline rule "twtag" to recognize widget tags ///
+
+var TWTagRegEx = /<\$[A-Za-z][A-Za-z0-9]*(?:\s+[^>]+)?>|<\/\$[A-Za-z][A-Za-z0-9]*\s*>/gm;
+
+function twtag_inline(state,silent) {
+	var ch, match, max, token, pos = state.pos;
+
+	if(!state.md.options.html) { return false; }
+
+	max = state.posMax;
+	if(state.src.charCodeAt(pos) !== 0x3C/* < */ ||
+		pos + 2 >= max) {
+		return false;
+	}
+
+	// Quick fail on second char
+	ch = state.src.charCodeAt(pos + 1);
+	if(ch !== 0x24/* $ */ && ch !== 0x2F/* / */) {
+		return false;
+	}
+
+	TWTagRegEx.lastIndex = pos;
+	match = TWTagRegEx.exec(state.src);
+	if(!match) { return false; }
+
+	if(!silent) {
+		token = state.push('twtag','',0);
+		token.content = state.src.slice(pos,pos + match[0].length);
+	}
+	state.pos += match[0].length;
+	return true;
+};
+
+function render_twtag(tokens,idx/*,options,env*/) {
+	return tokens[idx].content;
+};
+
+/// Add inline rule "macrocall" to parse <<macroname ...>> ///
+
+var MacroCallRegEx = /\s*<<([^\s>"'=]+)[^>]*>>/g;
+function macrocall_inline(state,silent) {
+	var match, max, pos = state.pos;
+
+	if(!state.md.options.html) { return false; }
+
+	// Check start
+	max = state.posMax;
+	if(state.src.charCodeAt(pos) !== 0x3C && state.src.charCodeAt(pos+1) !== 0x3C /* << */||
+		pos + 3 >= max) {
+	  return false;
+	}
+
+	MacroCallRegEx.lastIndex = pos;
+	match = MacroCallRegEx.exec(state.src);
+	if(!match) { return false; }
+
+	if(!silent) {
+		var token = state.push('macrocall','',0);
+		token.content = state.src.slice(pos,pos+match[0].length);
+	}
+	state.pos += match[0].length;
+	return true;
+}
+
+function render_macrocall(tokens,idx) {
+	return tokens[idx].content;
+}
+
+/// post processing ///
+
+function wikify(state) {
+	var href, title, src, alt, replaceTag = undefined;
+
+	state.tokens.forEach(function(blockToken) {
+		if(blockToken.type === 'inline' && blockToken.children) {
+			for(let token of blockToken.children) {
```

Comment:
we don't use `let` at the moment

Reply:
That's correct - we use (roughly) ES5.1 to maintain compatibility with older browsers without using transpilation.

Reply:
fixed